power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -198,6 +198,21 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
                 update[entry.index](*this, found->second, pos, sequence_idx_map[entry.index]);
             }
         }
+
+        // sequence_idx_map accumulation logic
+        if (is_accumulated_component_updated_) { // assign sequence_idx_map
+            cached_sequence_idx_map_.assign(sequence_idx_map.begin(), sequence_idx_map.end());
+            is_accumulated_component_updated_ = false;
+        } else { // accumulate sequence_idx_map, take the union of each sequence_idx_map
+            for (size_t i = 0; i < n_types; ++i) {
+                for (const auto& elem : sequence_idx_map[i]) {
+                    if (std::find(cached_sequence_idx_map_[i].begin(), cached_sequence_idx_map_[i].end(), elem) ==
+                        cached_sequence_idx_map_[i].end()) {
+                        cached_sequence_idx_map_[i].push_back(elem);
+                    }
+                }
+            }
```

Comment:
i think you can just append all values here. it's fine if a component is added to this list multiple times. the accumulate logic in `y_bus` will already take care of duplicates. it makes this logic easier with only a slight memory overhead but computationally faster in the average case.

Reply:
I wouldn't disagree here --- almost wrote `set` but somewhere rang Tony's comment

Reply:
Changes incorporated in the newest commit. Conversation resolved.